notify/impl.go

```diff
@@ -678,12 +671,21 @@ func (n *Slack) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
 
 	var numFields = len(n.conf.Fields)
 	if numFields > 0 {
-		var fields = make([]slackAttachmentField, numFields)
-		for k, v := range n.conf.Fields {
-			fields[k] = slackAttachmentField{
-				tmplText(v["title"]),
-				tmplText(v["value"]),
-				n.conf.ShortFields,
+		var fields = make([]config.SlackField, numFields)
+		for index, field := range n.conf.Fields {
+			// Check if short was defined for the field otherwise fallback to the global setting
+			var short *bool
```

Comment:
nit but I would do:

```go
                        var short bool
			if field.Short != nil {
				short = *field.Short
			} else {
				short = n.conf.ShortFields
			}
```

Reply:
The main reason it is being set as it is currently is that I reused `SlackField` from the config. package, which has 'Short` defined as a `*bool` now. I initially had it like that, but changed it after the tests failed and caught the error. I can change the if statement to be as above, and then pass a pointer to the variable `short` when building the `SlackField`.